### ISSUE TITLE
Scientific notation does not work for all numeric inputs

### ISSUE THREAD

**Issue by btlvr** [gnuradio/gnuradio#2553]
In some blocks, the user has the ability to enter, for example, `32e3` as a shorthand for `32,000`. This feature does not seem to be available in all blocks/inputs. 

For example, I can supply `2e6` for `quadrature rate` in the `WMFM Receive` block, but not for the `interpolation` of the `Rational Resampler` block. 
This throws the following error:
```
Param - Interpolation(interp):
	Expression "32000.0" is invalid for type integer.
```

In my opinion, if the input requires an integer, the value should automatically be cast to an integer. *Especially* if the float can be converted to an integer without any loss of precision. It's very inconvenient to have to type several zeros.

**Comment by 777arc** [comment]
Yeah I have always found that annoying, but I would consider it a feature request.  Maybe once 3.8 is done we can get to some of these really good feature requests you suggested.
